contracts/invariant.ral

```diff
@@ -27,22 +48,25 @@ Contract Invariant(
         PriceLimitReached = 11
         InvalidProtocolFee = 12
         NotOwner = 13
+        ZeroAmount = 14
+        WrongLimit = 15
+        NoGainSwap = 16
+        PositionDoesNotExist = 17
     }
 
     @using(preapprovedAssets = false, updateFields = true)
     pub fn init() -> () {
         checkCaller!(callerAddress!() == config.admin, InvariantError.NotAdmin)
         assert!(!init, InvariantError.AlreadyInitialized)
         let invtId = selfAddress!()
-        let positionsId = positions.getAddress()
-        let swapId = swap.getSelfId()
 
         feeTiers.setAdmins(invtId)
         poolKeys.setAdmins(invtId)
-        pools.setAdmins(invtId, positionsId, swapId)
-        positions.setAdmins(invtId)
-        tickmap.setAdmins(invtId, swapId)
-        ticks.setAdmins(invtId, positionsId, swapId)
+        pools.setAdmins(invtId)
+        tickmap.setAdmins(invtId)
+        ticks.setAdmins(invtId)
+
+        PositionsCounter(positionsCounterContractId).setAdmins(selfAddress!())
```

Comment:
`selfAddress!()` is assigned few lines above to `invtId` there is no need to invoke macro once again